gnovm/pkg/gnolang/preprocess.go

```diff
@@ -1891,6 +1916,10 @@ func Preprocess(store Store, ctx BlockNode, n Node) Node {
 							} else {
 								last.Define(ln, anyValue(rt))
 							}
+							// if rhs is untyped
+							if isUntyped(rt) {
```

Comment:
In what test case is this necessary? Won't these already be converted to types above on line 1868?